### ISSUE TITLE
Add name to Hamburger menu on all pages

### ISSUE THREAD

**Issue by lhenze** [inveniosoftware/invenio-app-rdm#1196]
From NYU A11y review -

At various responsive break points, the header navigation links move into a hamburger menu. This menu does not have an accessible name.
